### PR TITLE
Enable sending only ansibleMetadata event

### PR DESCRIPTION
**Pull request by ganeshrn**
*  Enable sending only ansibleMetadata event if telemetry is enabled
*  Add python version in ansibleMetadata event payload
*  Store the ansibleMetadata event globally to send the event only once across user seessions if any of the parameters within the payload is changed
*  Additionally send the event only when there is change in the extension settings or a new Asible file is opened with different ansibleMetadata payload.

Related: https://issues.redhat.com/browse/AAP-24906

### CODE REVIEW

File: src/utils/telemetryUtils.ts [hunk 1]
```diff
@@ -134,10 +158,11 @@ export class TelemetryErrorHandler implements ErrorHandler {
   }
 
   error(error: Error, message: Message, count: number): ErrorHandlerResult {
-    this.telemetry.send({
-      name: "ansible.lsp.error",
-      properties: { jsonrpc: message.jsonrpc, error: error.message },
+    sendTelemetry(this.telemetry, true, "ansible.lsp.error", {
```
Comment (TamiTakamiya) [inline src/utils/telemetryUtils.ts:161]: The second parameter is set to `true`. Does it mean that this telemetry event can be sent even if the telemetry service is not initialized?

File: src/utils/telemetryUtils.ts [hunk 2]
```diff
@@ -214,9 +239,8 @@ export class TelemetryOutputChannel implements vscode.OutputChannel {
       }
       this.errors.push(value);
       const timeoutHandle = setTimeout(() => {
-        this.telemetry.send({
-          name: "ansible.server.error",
-          properties: { error: this.createErrorMessage() },
+        sendTelemetry(this.telemetry, true, "ansible.server.error", {
```
Comment (TamiTakamiya) [inline src/utils/telemetryUtils.ts:242]: Same as on line 161.

**Review by TamiTakamiya — COMMENTED** [review]
I left two comments (both on the same topic). Thanks.

File: src/utils/telemetryUtils.ts (hunk 1 above)
Comment (ganeshrn) [reply src/utils/telemetryUtils.ts:161]: Have added `sendTelemetry` function to provide a single to call the telemetry send API. In the earlier code TelemetryErrorHandler call we are invoking the send method directly without checking for initiliased however there is a check added to ensure telemetryService object is initialised.

Comment (TamiTakamiya) [reply src/utils/telemetryUtils.ts:161]: Thanks for the explanation!

**Review by TamiTakamiya — APPROVED** [review]

**Review by ssbarnea — APPROVED** [review]
LGTM, assuming that coverage does not go down.

**Pull request merged**